### PR TITLE
Add avatar preview feature

### PR DESCRIPTION
**Pull request by Puneet-here**
### Fixed Issues

$ https://github.com/Expensify/App/issues/11197
PROPOSAL: https://github.com/Expensify/App/issues/11197#issuecomment-1264004163

### Tests

1. Go to settings > profile 
2. Click on avatar and add a photo if you don't have any
3. Click on the avatar again 
4. Verify you see an option to view photo 
5. Verify you can see the avatar preview

---
1. Go to settings >worskpaces  
2. Click on a workspace and go to settings
3. Click on the avatar and add a photo if you don't have any
4. Click again and verify you see an option to view photo 
6. Verify you can see the avatar preview

- [ ] Verify that no errors appear in the JS console

### Offline tests

Same
### QA Steps

 Same as tests
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/101883770/c9082ed9-651b-426b-b218-ecd232a5f022

https://github.com/Expensify/App/assets/101883770/721941fe-cd8d-4f56-965e-1c5a755ca202

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/101883770/011dc202-7de5-44c1-a3d4-123fa59ad6ed

https://github.com/Expensify/App/assets/101883770/3871049b-a800-4049-8087-e09e49d43b54

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/101883770/97ecd739-0c36-47ee-a41f-2e169c4ccaf2

https://github.com/Expensify/App/assets/101883770/93b677bc-5e8c-4c35-9401-cc0cb97ebad3

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/101883770/274c2b89-f067-430f-9841-d5d70eb5df62

https://github.com/Expensify/App/assets/101883770/bd9f5296-fdec-4f19-8c98-46d4c5f02b86

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/101883770/acd33dd8-b50e-47a0-b992-19d8e2223ae3

https://github.com/Expensify/App/assets/101883770/65f6c9ee-79b8-4437-94ae-ad6855ac692a

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/101883770/e8e19907-174d-41d0-abce-b3a7f97dd290

https://github.com/Expensify/App/assets/101883770/545d1ca6-3e4b-4037-8e60-e7aa9d262da8

</details>

### CODE REVIEW

**Review by venture1981 — COMMENTED** [review]
@Puneet-here  seems like it works on windows, but not tested on Mac.

**Comment by Puneet-here** [comment]
@rushatgabhane, done!

File: src/components/AvatarWithImagePicker.js [hunk 1]
```diff
@@ -265,6 +279,11 @@ class AvatarWithImagePicker extends React.Component {
                     this.props.onImageRemoved();
                 },
             });
+            menuItems.unshift({
```
[the remark this answers is not in the document]
Comment (Puneet-here) [reply src/components/AvatarWithImagePicker.js:282]: We have 3 items: View photo, Upload photo and Remove photo. Remove and View photo items are shown conditionally (when the user isn't using the default avatar) so we only push those items in the array when the condition matches that's why I used unshift to move the view photo option at the top. https://github.com/Expensify/App/blob/e22931e003830f90c631f13edc5b9ac01a48eb7e/src/components/AvatarWithImagePicker.js#L261-L287

**Comment by rushatgabhane** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@Puneet-here bug: attachment preview doesn't show the image

1. upload profile photo
2. click view profile

Expected: attachment preview shows the uploaded photo
Actual: You see attachment icon only

https://github.com/Expensify/App/assets/29673073/2ce3c4a7-756d-4635-976e-9a00538929a9

**Comment by Puneet-here** [comment]
Is there any extra step. I wasn't able to repro it

https://github.com/Expensify/App/assets/101883770/9a0d2110-ccea-46fc-aec2-b3350302847a


https://github.com/Expensify/App/assets/101883770/5983f9f7-2e18-477b-aed2-3b2fdbfaf143

**Comment by Puneet-here** [comment]
> Is there any extra step. I wasn't able to repro it


bump @rushatgabhane

**Comment by rushatgabhane** [comment]
@Puneet-here no extra steps. But it's consistently reproducible.

**Comment by Puneet-here** [comment]
I was finally able to reproduce it. So it happens when you open it immediately. You have to be super fast. Looking into the problem.

**Comment by rushatgabhane** [comment]
maybe being offline will consistently reproduce it?

**Comment by Puneet-here** [comment]
> maybe being offline will consistently reproduce it?

I tried a slow internet connection and also being offline but it didn't work. You just have to be very fast to repro this. I just tried checking the avatar preview in user profile page and it seems like this is reproducible there as well so I can confirm that this behaviour is already present in our attachment modal and not related to this PR. I think we should open a new ticket for this. It's really hard to repro on user profile page so here's how I would suggest reproducing it-

1. Login to your account on two devices 
2. In device 1 go to any room and go to members 
3. Click on your own profile to see the user details
4. At device 2 go to profile page and upload an image 
5. Now on device 1 click on the avatar asap

**Comment by Puneet-here** [comment]
bump @rushatgabhane

**Comment by rushatgabhane** [comment]
@Puneet-here could you please resolve conflicts. thx!

**Comment by Puneet-here** [comment]
Done @rushatgabhane

**Comment by rushatgabhane** [comment]
Looks like we had more conflicts over the weekend 😅

**Comment by Puneet-here** [comment]
Done @rushatgabhane

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
**Bug**: mWeb safari - native attachment picker opens on clicking view photo

https://github.com/Expensify/App/assets/29673073/ec7ec124-9904-47b3-95d9-6d8b870edb0b

**Comment by Puneet-here** [comment]
@rushatgabhane, I am not sure what the problem was but i found that it only happens when view photo option is on top. I have changed its position to last. Hope it's fine.

**Comment by rushatgabhane** [comment]
@Puneet-here thanks! I think we need to dive deeper. I think `view photo` option should be the first item in the menu.

**Comment by Puneet-here** [comment]
@rushatgabhane, maybe we should get confirmation from design team.

**Comment by Puneet-here** [comment]
@rushatgabhane, I believe we can move forward with the current change since others don't think strongly about keeping the view photo as the first option?

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@Puneet-here the merge freeze is officially over! Let's merge with latest `main` an re-test once, thanks!

**Comment by Puneet-here** [comment]
Done @rushatgabhane

**Review by rushatgabhane — APPROVED** [review]
@stitesExpensify LGTM!

checklist - https://github.com/Expensify/App/pull/22145#issuecomment-1633523631

**Review by stitesExpensify — APPROVED** [review]
LGTM

**Comment by ayazhussain79** [comment]
Should the "Preview" header for "Profile avatar" be updated after uploading an image to "Profile picture"?




Screenshot 2023-09-19 at 4 31 40 PM

**Pull request merged**